File: src/algorithm/search/linear-search/linear_search.py
#! /usr/local/bin/python3


class LinearSearch:
    def search(self, element, search_list):
        if search_list is None:
            return -1

        for i, val in enumerate(search_list):
            if val == element:
                return i

        return -1

    def test(self):
        import time

        start = time.process_time()
        result = self.search(4, [3, 5, 7, 33, 4, 6, 9, 8, 3])
        end = time.process_time()
        print(result, (end - start) * 1000)

if __name__ == '__main__':
    LinearSearch().test()

File: src/algorithm/search/linear-search/linearSearch.js
#! /usr/local/bin/node

class LinearSearch {
    search(element, list=[]) {
        for (let i = 0; i < list.length; i++) {
            if (list[i] === element) {
                return i;
            }
        }

        return -1;
    }

    test() {
        console.time('search');
        console.log(this.search(4, [3, 5, 7, 33, 6, 9, 8, 3]));
        console.timeEnd('search');
    }
}

new LinearSearch().test();
